internal/compiler-interface/src/main/java/xsbti/compile/JavaTool.java

```diff
@@ -0,0 +1,26 @@
+package xsbti.compile;
+
+import java.io.File;
+import xsbti.Logger;
+import xsbti.Reporter;
+
+/**
+ * An interface for on of the tools in the java tool chain.
+ *
+ * We assume the following is true of tools:
+ * - The all take sources and options and log error messages
+ * - They return success or failure.
+ */
```

Comment:
It looks like they also take a reporter. But also, why duplicate this info here?